File: app/src/types/topic.ts
export interface Topic {
  id: number
  title: string
  user_id: number
  username?: string
  score: number
  created_at: string
  updated_at: string
}

export interface CreateTopicRequest {
  title: string
  user_id: number
}

export interface UpdateTopicRequest {
  title?: string
}

File: internal/models/topic.go
package models

import "time"

type Topic struct {
	ID        int64     `json:"id"`
	Title     string    `json:"title"`
	UserID    int64     `json:"user_id"`
	Username  string    `json:"username,omitempty"`
	CreatedAt time.Time `json:"created_at"`
	UpdatedAt time.Time `json:"updated_at"`
	Score     int64     `json:"score"`
}

type CreateTopicRequest struct {
	Title  string `json:"title" binding:"required,min=5,max=200"`
	UserID int64  `json:"user_id" binding:"required"`
}

type UpdateTopicRequest struct {
	Title string `json:"title" binding:"omitempty,min=5,max=200"`
}
